4337-gas-metering/pimlico/pimlico.ts

```diff
@@ -0,0 +1,275 @@
+const erc721TokenAddress = process.env
+  .PIMLICO_ERC721_TOKEN_CONTRACT as `0x${string}`;
+
+const argv = process.argv.slice(2);
+if (argv.length != 1) {
+  throw new Error("TX Type Argument not passed."); // account || erc20 || erc721
+}
+
+const txType: string = argv[0];
+if (txType != "account" && txType != "erc20" && txType != "erc721") {
+  throw new Error("TX Type Argument Invalid");
+}
+
+const safeAddresses = (
+  SAFE_ADDRESSES_MAP as Record<string, Record<string, any>>
+)[safeVersion];
+let chainAddresses;
+if (safeAddresses) {
+  chainAddresses = safeAddresses[chainID];
+}
+
+if (apiKey === undefined) {
+  throw new Error(
+    "Please replace the `apiKey` env variable with your Pimlico API key",
+  );
+}
+
+if (!privateKey) {
+  throw new Error(
+    "Please populate .env file with demo Private Key. Recommended to not use your personal private key.",
+  );
+}
+
+const signer = privateKeyToAccount(privateKey as Hash);
+console.log("Signer Extracted from Private Key.");
+
+let bundlerClient;
+let publicClient;
+if (chain == "goerli") {
+  bundlerClient = createClient({
+    transport: http(`https://api.pimlico.io/v1/${chain}/rpc?apikey=${apiKey}`),
+    chain: goerli,
+  })
+    .extend(bundlerActions)
+    .extend(pimlicoBundlerActions);
+
+  publicClient = createPublicClient({
+    transport: http(rpcURL),
+    chain: goerli,
+  });
+} else {
+  throw new Error(
+    "Current code only support limited networks. Please make required changes if you want to use custom network.",
+  );
+}
+
+const initCode = await getAccountInitCode({
+  owner: signer.address,
+  addModuleLibAddress: chainAddresses.ADD_MODULES_LIB_ADDRESS,
+  safe4337ModuleAddress: chainAddresses.SAFE_4337_MODULE_ADDRESS,
+  safeProxyFactoryAddress: chainAddresses.SAFE_PROXY_FACTORY_ADDRESS,
+  safeSingletonAddress: chainAddresses.SAFE_SINGLETON_ADDRESS,
+  saltNonce: saltNonce,
+  multiSendAddress: multiSendAddress,
+  erc20TokenAddress: usdcTokenAddress,
+  paymasterAddress: erc20PaymasterAddress,
+});
+console.log("\nInit Code Created.");
+
+const senderAddress = await getAccountAddress({
+  client: publicClient,
+  owner: signer.address,
+  addModuleLibAddress: chainAddresses.ADD_MODULES_LIB_ADDRESS,
+  safe4337ModuleAddress: chainAddresses.SAFE_4337_MODULE_ADDRESS,
+  safeProxyFactoryAddress: chainAddresses.SAFE_PROXY_FACTORY_ADDRESS,
+  safeSingletonAddress: chainAddresses.SAFE_SINGLETON_ADDRESS,
+  saltNonce: saltNonce,
+  multiSendAddress: multiSendAddress,
+  erc20TokenAddress: usdcTokenAddress,
+  paymasterAddress: erc20PaymasterAddress,
+});
+console.log("\nCounterfactual Sender Address Created:", senderAddress);
+console.log(
+  "Address Link: https://" + chain + ".etherscan.io/address/" + senderAddress,
+);
+
+const contractCode = await publicClient.getBytecode({ address: senderAddress });
+
+if (contractCode) {
+  console.log("\nThe Safe is already deployed.");
+  if (txType == "account") {
+    console.log("");
```

Comment:
Is this intentional?

Reply:
I am currently working on the native transfer one as well. Once that is done, I will try to take away the functions which are similar in both the services.

As this is being worked on, should I move it back to "In Progress"?

Reply:
Imo, "In progress" works better here.